x/ibc/20-transfer/module.go

```diff
@@ -324,7 +324,7 @@ func (am AppModule) OnTimeoutPacket(
 	packet channeltypes.Packet,
 ) (*sdk.Result, error) {
 	var data FungibleTokenPacketData
-	if err := types.ModuleCdc.UnmarshalBinaryBare(packet.GetData(), &data); err != nil {
+	if err := types.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
```

Comment:
is this a bug?

Reply:
We just need to marshal & unmarshal in the same way; at the moment I think we marshal as JSON.